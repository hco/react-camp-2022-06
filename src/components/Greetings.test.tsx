import { render, screen } from "@testing-library/react";
import Greeting from "./Greetings";

describe("<Greeting />", () => {
  test("it renders without errors", () => {
    render(<Greeting name="Danny" />);
  });

  test("it mentions the given person", () => {
    render(<Greeting name="Danny" />);

    screen.getByText("danny", { exact: false });
    screen.getByText(/danny/i);
  });

  test("it renders a paragraph with a greeting", () => {
    const { container } = render(<Greeting name="Danny" />);

    // Das hier entspricht nicht den best-practices die von der
    // react-testing-library empfohlen warden!!
    // eslint-disable-next-line testing-library/no-node-access
    expect(container?.firstChild?.nodeName).toBe("P");

    // eslint-disable-next-line testing-library/no-node-access
    expect(container?.firstChild).toHaveTextContent("Hello, Danny!");
  });
});
