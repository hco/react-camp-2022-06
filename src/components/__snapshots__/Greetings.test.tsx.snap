// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Greeting /> it renders according to snapshot 1`] = `
<div>
  <p>
    Hello, 
    Danny
    !
  </p>
</div>
`;

exports[`<Greeting /> it renders according to snapshot: 
      <div>
        <p>
          Hello, 
          Danny
          !
        </p>
      </div>
     1`] = `
<div>
  <p>
    Hello, 
    Danny
    !
  </p>
</div>
`;
